force-app/main/default/lwc/geFormRenderer/geFormRendererHelper.js

```diff
@@ -31,15 +31,15 @@ export function flatten(obj) {
  */
 export function convertBDIToWidgetJson(additionalObjectJson) {
     if (isEmptyObject(additionalObjectJson)) {
-        return;
+        return additionalObjectJson;
     }
 
     const additionalObjects = JSON.parse(additionalObjectJson);
 
     if (isEmptyObject(additionalObjects) ||
         !additionalObjects.hasOwnProperty('dynamicSourceByObjMappingDevName')) {
 
-        return;
+        return additionalObjectJson;
```

Comment:
Great questions!

1. Yep, this will need to be cleared out once we've got allocations as a type as well and only need to perform such transformations in apex.

2. When we return early in this method on line 42, it means that the additional object json has already been transformed to the format that the allocations widget can consume. I can make an update to return a better-named variable for this particular scenario. Thanks for catching!